v0/destinations/serenytics/transform.js

```diff
@@ -0,0 +1,208 @@
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  getHashFromArrayWithDuplicate,
+  ErrorMessage
+} = require("../../util");
+
+const {
+  CONFIG_CATEGORIES,
+  SERENYTICS_TRACK_EXCLUSION_LIST,
+  SERENYTICS_IDENTIFY_EXCLUSION_LIST,
+  SERENYTICS_PAGE_SCREEN_EXCLUSION_LIST
+} = require("./config");
+const {
+  payloadBuilder,
+  storageUrlResponseBuilder,
+  responseBuilder
+} = require("./utils");
+
+const trackResponseBuilder = (message, { Config }, payload) => {
+  const STORAGE_URL = Config.storageUrlTrack || "";
+  const storageUrlEventMapping = getHashFromArrayWithDuplicate(
+    Config.eventToStorageUrlMap,
+    "from",
+    "to",
+    false
+  );
+  const { event } = message;
+  if (!event) {
+    throw new CustomError(
+      `[Serenytics]: event name is required in track call.`,
+      400
+    );
+  }
+  if (!storageUrlEventMapping[event] && !STORAGE_URL) {
+    throw new CustomError(
+      `[Serenytics]: storage url for Track is required.`,
+      400
+    );
+  }
+
+  const storageUrlEventList = storageUrlEventMapping[event];
+  if (storageUrlEventList) {
+    const responseList = storageUrlResponseBuilder(
+      STORAGE_URL,
+      storageUrlEventList,
+      payload
+    );
+    return responseList;
```

Comment:
Apart from pushing the response with the default track URL and returning from here, we can create a responseList array and push all the responses there.

Reply:
updated